ledger-tool/src/main.rs

```diff
@@ -1027,6 +1027,10 @@ fn main() {
 
                     println!("Creating a snapshot of slot {}", bank.slot());
                     bank.squash();
+                    for slot in 0..=(snapshot_slot - 1000) {
```

Comment:
Why `-1000`?   But could we just call `bank.compact_stale_slots()` instead?

Reply:
1000 is arbitrarily chosen. I was in haste. This pr is wip to show this acutally solves #9196. Come to think of it, snapshot don't have non-rooted slots; so no need for substraction.

bank.compact_stale_slots() does take consider of `non-rooted` slots. And, it assumes constant update of `uncleaned_roots`?

Reply:
This is fixed by cleaning up!